index.bs

```diff
@@ -411,49 +411,89 @@ Note: The high-entropy portions of the user agent information are retrieved thro
-        with [=user agent/brand=] as {{NavigatorUABrandVersion/brand}} and [=user agent/significant version=] as {{NavigatorUABrandVersion/version}}.
+        with [=user agent/brand=] as {{NavigatorUABrandVersion/brand}} and [=user agent/significant version=] as
+        {{NavigatorUABrandVersion/version}}.
 
     2. Append |dict| to |list|.
 
 4.  The [=user agent=] SHOULD execute the following steps:
 
-    1.  [=list/Append=] additional items to |list| containing {{NavigatorUABrandVersion}} objects,
-        initialized with arbitrary {{NavigatorUABrandVersion/brand}} and {{NavigatorUABrandVersion/version}} combinations.
+    1.  [=list/Append=] one additional [=list/item=] to |list| containing a {{NavigatorUABrandVersion}} dictionary,
+        initialized with <a lt="create an arbitrary brand">arbitrary brand</a> and <a lt="create an arbitrary version">
+        arbitrary version</a> combinations.
 
-    2.  Randomize the order of the items in |list|.
+    2.  Randomize the order of the [=list/items=] in |list|.
 
     Note: One approach to minimize caching variance when generating these random components could be to
     determine them at build time, and keep them identical throughout the lifetime of the [=user agent=]'s significant
-    version. 
+    version.
 
     Note: See [[#grease]] for more details on when and why these randomization steps might be appropriate.
 
 5. Return |list|.
 
-<h3 id="getters">Getters</h3>
+An <dfn for="user agent" export>equivalence class</dfn> represents a group of browsers believed to be compatibile with
+each other. A shared rendering engine may form an [=equivalence class=], for example.
+
+<h4 id="create-arbitrary-brands-section"
+    algorithm="to create arbitrary brand and version values">Create arbitrary brand and version values</h4>
+
+To <dfn>create an arbitrary brand</dfn>, the [=user agent=] MUST run these steps:
+
+    1. Let |arbitraryBrand| be a [=/string=] composed of [=ASCII lower alpha=], [=ASCII upper alpha=], or 0x20 (SP), no
+         longer than twenty [=ASCII bytes=].
+    1. Let |arbitraryBrandList| be the result of <a lt="split on ASCII whitespace">splitting |arbitraryBrand| on ASCII
+         whitespace</a>.
+    1. Let |greaseyStack| be a [=stack=].
+    1. Let |greaseyChars| be the [=list=] of [=ASCII bytes=] « 0x20 (SP), 0x28 (left parenthesis), 0x29
+         (right parenthesis), 0x2D (-), 0x2E (.), 0x2F (/), 0x3A (:), 0x3B (;), 0x3D (=), 0x3F (?), 0x5F (_) ».
+    1. For each item of |arbitraryBrandList|, [=stack/push=] a randomly selected [=list/item=] from |greaseyChars| onto
+         |greaseyStack|.
+    1. Let |greaseyBrandList| be a [=list=] and |index| be 0.
+    1. While |greaseyStack| [=stack/is not empty=]:
+        1. Let |item| be the result of <a lt="pop">popping</a> from |greaseyStack|.
+        1. Append |item| to |greaseyBrandList|.
+        1. Append |arbitraryBrandList|[|index|] to |greaseyBrandList|.
+        1. Increment |index| by 1.
```

Comment:
Might this end up in an array-out-of-bounds?

Reply:
I don't see how, do you have a scenario where it might?

Reply:
I misread step 5; silly me